pkg/checks/checks.go

```diff
@@ -0,0 +1,167 @@
+package checks
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"syscall"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	appclient "k8s.io/client-go/kubernetes/typed/apps/v1"
+)
+
+// WaitForHTTPStatusOk poll until URL returns 200
+func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urltocheck string) error {
+	cli := &http.Client{
+		Transport: transport,
+		Timeout:   10 * time.Second,
+	}
+
+	req, err := http.NewRequest("GET", urltocheck, nil)
+	if err != nil {
+		return err
+	}
+	return wait.PollInfinite(time.Second, func() (bool, error) {
```

Comment:
sorry, my bad, should be `wait.PollUntil(time.Second, func() (bool, error) { ... }, ctx.Done())`